rostime/src/time.cpp

```diff
@@ -167,6 +167,40 @@ namespace ros
     nsec = nsec_sum;
 #endif
   }
+
+  void ros_steadytime(uint32_t& sec, uint32_t& nsec)
+#ifndef WIN32
+    throw(NoHighPerformanceTimersException)
+#endif
+  {
+#ifndef WIN32
+    timespec start;
+    clock_gettime(CLOCK_MONOTONIC, &start);
+    sec  = start.tv_sec;
+    nsec = start.tv_nsec;
+#else
+    static LARGE_INTEGER cpu_frequency, performance_count;
```

Comment:
Why should these variables be static? A local variable would avoid problems of the static variable being used by multiple concurrent callers.